.github/workflows/publish-go-nightly-task.yml

```diff
@@ -144,9 +157,6 @@ jobs:
           tar -czvf "$PACKAGE_FILENAME" \
           -C "${{ env.PROJECT_NAME }}_osx_${{ matrix.artifact.name }}/" "${{ env.PROJECT_NAME }}" \
           -C ../../ LICENSE.txt
-          CHECKSUM_LINE="$(shasum -a 256 $PACKAGE_FILENAME)"
-          echo "PACKAGE_FILENAME=$PACKAGE_FILENAME" >> $GITHUB_ENV
```

Comment:
`PACKAGE_FILENAME` was used in the step below for a reason: we used to upload only the notarized binary. This way [we are overwriting all the binaries](https://github.com/MatteoPologruto/arduino-cli/actions/runs/3105467121/jobs/5031149531#step:9:13): which means that either the arm64 or the amd64 one is not notarized anymore

Reply:
One way to enhance this further and remove completely this kind of errors would be to only download the artifact we are going to notarize and not all of them [like we do now](https://github.com/arduino/arduino-cli/pull/1883/files#diff-b01c8ff87205e83918583aff246152ef27d286cccaa14409dddc17bd37104518R83-R87). And then upload only the notarized one [here](https://github.com/arduino/arduino-cli/pull/1883/files#diff-b01c8ff87205e83918583aff246152ef27d286cccaa14409dddc17bd37104518R166)

Reply:
Unfortunately, it is not possible to download the single artifact to be notarized since only one artifact named `dist` is uploaded in the previous job. One way to do it could be to upload every artifact generated in the building process separately, but it would mean that they'll need to be treated individually when calculating the checksum. I chose to simply reintroduce the above mentioned line.